src/main/java/frc/robot/subsystems/LEDManager.java

```diff
@@ -26,34 +29,69 @@
  */
 public class LEDManager {
 
+    private static LEDManager instance;
+
     // Hardware
     private AddressableLED ledStrip;
     private AddressableLEDBuffer ledBuffer;
 
     /**
-     * Creates a new LEDManager object using the provided PWM port number.
+     * returns the singleton object of the LED Manager
+     * @return the singleton object
+     */
+    public static LEDManager getInstance() {
+        if (instance == null) {
+            throw new IllegalStateException("LEDManager has not been initialized. Call initialize() first.");
+        } else {
+            return instance;
+        }
+    }
+
+    /**
+     * initialized the LEDManager object using the provided PWM port number and length.
+     * 
+     * @param port The port number the LED strip is plugged into on the PWM.
+     * @param length The number of LEDs in the strip
+     */
+    public static void initialize(int port, int length) {
+        if (instance != null) {
+            throw new IllegalStateException("LEDManager has already been initialized.");
+        }
+        instance = new LEDManager(port, length);
+    }
+
+    /**
+     * Creates a new LEDManager object using the provided PWM port number and length.
      * 
-     * @param portNum The port number the LED strip is plugged into on the PWM.
+     * @param port The port number the LED strip is plugged into on the PWM.
+     * @param length The number of LEDs in the strip
      */
-    public LEDManager(int portNum) {
+    private LEDManager(int port, int length) {
 
         // Create new AddressableLED and AddressableLEDBuffer objects.
-        this.ledStrip = new AddressableLED(portNum);
-        this.ledBuffer = new AddressableLEDBuffer(60); // TODO: Figure out actual number. 60 is just a placeholder.
+        this.ledStrip = new AddressableLED(port);
+        this.ledBuffer = new AddressableLEDBuffer(length);
         
         // Setup the LED strip hardware
         this.ledStrip.setLength(this.ledBuffer.getLength());
         this.ledStrip.setData(ledBuffer);
         this.ledStrip.start();
-    } 
+    }
 
     /**
-     * To set ledPattern, method taking in a LEDConfiguration value.
-     * 
-     * @param ledConfiguration The LEDConfiguration this LEDManager's LED strip will display.
+     * creates a view to control a segment of the led strip
+     * @param ledSegment the segment to create the view out of
+     * @return the view to control the segment
      */
-    public void LEDPatternSet(LEDConfiguration ledConfiguration) {
-        ledConfiguration.getLEDPattern().applyTo(this.ledBuffer);
-        this.ledStrip.setData(this.ledBuffer);
+    public static AddressableLEDBufferView getViewForSegment(LEDSegment ledSegment) {
+        return instance.ledBuffer.createView(ledSegment.getStartingIndex(),ledSegment.getEndingIndex());
```

Comment:
```suggestion
        return instance.ledBuffer.createView(ledSegment.getStartingIndex(), ledSegment.getEndingIndex());
```
Reminder on space after comma :P